src/net/tls.cc

```diff
@@ -1358,6 +1358,12 @@ class session : public enable_lw_shared_from_this<session> {
             gnutls_transport_set_errno(*this, EAGAIN);
             return -1;
         }
+        if (_output_pending.failed()) {
+            // copy exception for on_internal_error
```

Comment:
How can you even get into a situation like this? The only way I can think of is if somehow a `put` managed to both generate a write that completed in exception _and_ caused a gnutls error other than `EAGAIN` to be returned?

IIRC, the only other path that does writing is handshake, and that always waits for output. But note that `handle_output_error` always ensures _output_pending is resolved before throwing the actual `_error`.